### ISSUE TITLE
[Bug]: Version workflow to auto update input manifests is failing

### ISSUE THREAD

**Issue by zelinh** [opensearch-project/opensearch-build#2854]
### Describe the bug

Our version workflow to auto raise PR for input manifest update is currently failing for almost 10 days. https://github.com/opensearch-project/opensearch-build/actions/workflows/versions.yml
The context of error:
```
BUILD FAILED in 11s
2022-11-04 00:49:45 WARNING  Error getting version of performance-analyzer: Command './gradlew properties -Dopensearch.version=3.0.0 -Dbuild.snapshot=false' returned non-zero exit status 1., ignored
2022-11-04 00:49:45 INFO     Checking out opensearch-observability#main
2022-11-04 00:49:45 INFO     Executing "git init" in /tmp/tmpco2c3n05/opensearch-observability
2022-11-04 00:49:45 INFO     Executing "git remote add origin https://github.com/opensearch-project/opensearch-observability.git" in /tmp/tmpco2c3n05/opensearch-observability
2022-11-04 00:49:45 INFO     Executing "git fetch --depth 1 origin main" in /tmp/tmpco2c3n05/opensearch-observability
2022-11-04 00:49:45 INFO     Removing /tmp/tmpco2c3n05
```
The problem might be the workflow trying to clone `https://github.com/opensearch-project/opensearch-observability.git` but the correct URL should be `https://github.com/opensearch-project/observability.git`
### To reproduce

https://github.com/opensearch-project/opensearch-build/actions/workflows/versions.yml

**Comment by mch2** [comment]
I don't think these maven issues are the reason for the wf failure. 

The [latest](https://github.com/opensearch-project/opensearch-build/actions/runs/3390231242/jobs/5634121899) failure has a separate issue at the bottom.  Where it is trying to fetch `https://github.com/opensearch-project/opensearch-observability` where the repo name is actually https://github.com/opensearch-project/observability.

The latest passing wf run still has those maven errors in it.  So those maven issues are a separate problem (looks like its trying to fetch a "released" version of 3.0 when 3.0 has not yet been released).

**Comment by zelinh** [comment]
> I don't think these maven issues are the reason for the wf failure.
> 
> The [latest](https://github.com/opensearch-project/opensearch-build/actions/runs/3390231242/jobs/5634121899) failure has a separate issue at the bottom. Where it is trying to fetch `https://github.com/opensearch-project/opensearch-observability` where the repo name is actually https://github.com/opensearch-project/observability.
> 
> The latest passing wf run still has those maven errors in it. So those maven issues are a separate problem (looks like its trying to fetch a "released" version of 3.0 when 3.0 has not yet been released).

Yes. I just updated the issue description. Thanks for the input here!

**Comment by mch2** [comment]
Looks like the repo name being cloned is tied to the name field in the manifest. This PR has a name of opensearch-observability when it needs to be observability.  This feels like a bug where name should not be driving the repo path.

**Comment by zelinh** [comment]
> Looks like the repo name being cloned is tied to the name field in the manifest. This [PR](https://github.com/opensearch-project/opensearch-build/commit/20dc58b38c38157aa63dbe88a59038aa433e5556) has a name of opensearch-observability when it needs to be observability. This feels like a bug where name should not be driving the repo path.

But seems like we have been using `opensearch-observability` as the name for previous manifest. 
e.g. https://github.com/opensearch-project/opensearch-build/blob/a872a391df1a02802425ad200ef273d66f4c6700/manifests/2.2.0/opensearch-2.2.0.yml#L135

**Comment by prudhvigodithi** [comment]
I see whats happening, looks to me like this a [bug](https://github.com/opensearch-project/opensearch-build/blob/main/src/manifests_workflow/component_opensearch.py#L29) and never shown up, the manifest update used to happen even before all components are loaded to the latest version manifest, here in this case its [3.0.0](https://github.com/opensearch-project/opensearch-build/tree/main/manifests/3.0.0), components like [opensearch-obervability](https://github.com/opensearch-project/opensearch-build/blob/main/manifests/3.0.0/opensearch-3.0.0.yml#L68-L74) or notifications-core, when added which has different name than the existing repo, this workflow `./manifests.sh update` would fail.
Till now this scenario did not come up, the `./manifests.sh update` would be successfully executed even before all components are loaded to the latest manifest
Log: `2022-11-05 00:10:19 INFO     Examining components in the latest manifest of OpenSearch (3.0.0)`
Now since for 3.0.0 the components like `opensearch-obervability` are added early hence this bug has shown up.

**Comment by zelinh** [comment]
> I see whats happening, looks to me like this a [bug](https://github.com/opensearch-project/opensearch-build/blob/main/src/manifests_workflow/component_opensearch.py#L29) and never shown up, the manifest update used to happen even before all components are loaded to the latest version manifest, here in this case its [3.0.0](https://github.com/opensearch-project/opensearch-build/tree/main/manifests/3.0.0), components like [opensearch-obervability](https://github.com/opensearch-project/opensearch-build/blob/main/manifests/3.0.0/opensearch-3.0.0.yml#L68-L74) or notifications-core, when added which has different name than the existing repo ([observability](https://github.com/opensearch-project/observability)), this workflow `./manifests.sh update` would fail. Till now this scenario did not come up, the `./manifests.sh update` would be successfully executed even before all components are loaded to the latest manifest Log: `2022-11-05 00:10:19 INFO Examining components in the latest manifest of OpenSearch (3.0.0)` Now since for 3.0.0 the components like `opensearch-obervability` are added early hence this bug has shown up.
> 
> Temporary fix: On my local for 3.0.0 input manifest I have updated the `obervability` as:
> 
> ```
>   - name: observability
>     repository: https://github.com/opensearch-project/observability.git
>     ref: main
>     platforms:
>       - linux
>       - windows
>     working_directory: opensearch-observability
> ```

Yes. I think this is the cause. 
https://github.com/opensearch-project/opensearch-build/blob/a872a391df1a02802425ad200ef273d66f4c6700/src/manifests_workflow/component_opensearch.py#L29)
it takes the name to construct the repo link instead of the repo link from the manifest. 
Another option should be updating the logic here to take the repository link from the existing manifest.

**Comment by rishabh6788** [comment]
Since we are already reading the manifest file, makes more sense to just use the repo field to fetch the component github url to avoid this situation in future. 
I will work on the fix. @prudhvigodithi @zelinh @mch2

**Comment by gaiksaya** [comment]
Fixed via #2889
